### PR TITLE
vcd_catalog resource allows to publish to external VCD organizations

### PR DESCRIPTION
**Pull request by vbauzysvmware**
Dep: https://github.com/vmware/go-vcloud-director/pull/424

allows configuring resource `vcd_catalog` to publish a catalog externally to make its vApp templates and media files available for subscription by organizations outside the Cloud Director installation.

new properties:
* publish_enabled
* cache_enabled
* preserve_identity_information
* password
* 
Organization needed settings will be enabled with config_gen to allow publishing catalogs externaly to run tests successfully.

### CODE REVIEW

File: vcd/datasource_vcd_catalog.go [hunk 1]
```diff
@@ -40,6 +41,21 @@ func datasourceVcdCatalog() *schema.Resource {
 				Optional: true,
 				Computed: true,
 			},
+			"publish_enabled": &schema.Schema{
+				Type:        schema.TypeBool,
+				Computed:    true,
+				Description: "True enables external publication",
```
Comment (lvirbalas) [inline vcd/datasource_vcd_catalog.go:47]: Would be nice if you could add more explanation of publication to where (same ask for the same line in resource down below).

File: website/docs/d/catalog.html.markdown [hunk 2]
```diff
@@ -43,6 +43,9 @@ The following arguments are supported:
 ## Attribute Reference
 
 * `description` - Catalog description.
+* `publish_enabled` - (*v3.6+*) Enable external publication. Default is `false`.
```
Comment (lvirbalas) [inline website/docs/d/catalog.html.markdown:46]: Here we definitely need more explanation of publication to where. Not all users know this feature.
Same ask for the resource part below.

**Review by lvirbalas — CHANGES_REQUESTED** [review]

File: vcd/datasource_vcd_catalog.go (hunk 1 above)
Comment (vbauzysvmware) [reply vcd/datasource_vcd_catalog.go:47]: It's name and explanation from VCD, not mine. Yet overall means that this catalog will be accessible for subscribe from different VCD.

Comment (lvirbalas) [reply vcd/datasource_vcd_catalog.go:47]: Once you extend the description in `catalog.html.markdown`, would help someone if you paste it here as well.

Comment (vbauzysvmware) [reply vcd/datasource_vcd_catalog.go:47]: I am not keen on explaining functionality I am not expert, but will try to take something usable from docs.

File: .changes/v3.6.0/772-features.md [hunk 3]
```diff
@@ -0,0 +1 @@
+* `vcd_catalog` allows configuring publish to external organizations [GH-772]
```
[the remark this answers is not in the document]
Comment (vbauzysvmware) [reply .changes/v3.6.0/772-features.md:1]: fixed

File: vcd/datasource_vcd_catalog.go (hunk 1 above)
Comment (vbauzysvmware) [reply vcd/datasource_vcd_catalog.go]: changed

File: website/docs/r/catalog.html.markdown [hunk 4]
```diff
@@ -53,11 +53,15 @@ The following arguments are supported:
...
+* `storage_profile_id` - (Optional, *v3.1+*) Allows to set specific storage profile to be used for catalog. **Note.** Data
 source [vcd_storage_profile](/providers/vmware/vcd/latest/docs/data-sources/storage_profile) can help to lookup storage profile ID.
-* `delete_recursive` - (Required) - When destroying use delete_recursive=True to remove the catalog and any objects it contains that are in a state that normally allows removal
-* `delete_force` -(Required) - When destroying use delete_force=True with delete_recursive=True to remove a catalog and any objects it contains, regardless of their state
+* `delete_recursive` - (Required) When destroying use delete_recursive=True to remove the catalog and any objects it contains that are in a state that normally allows removal
+* `delete_force` -(Required) When destroying use delete_force=True with delete_recursive=True to remove a catalog and any objects it contains, regardless of their state
+* `publish_enabled` - (Optional, *v3.6+*)  Enable allows to publish a catalog externally to make its vApp templates and media files available for subscription by organizations outside the Cloud Director installation. Default is `false`.
```
Comment (lvirbalas) [inline website/docs/r/catalog.html.markdown:61]: You have added double white space at the beginning.

**Review by lvirbalas — CHANGES_REQUESTED** [review]

File: website/docs/d/catalog.html.markdown [hunk 5]
```diff
@@ -43,6 +43,9 @@ The following arguments are supported:
 ## Attribute Reference
 
 * `description` - Catalog description.
+* `publish_enabled` - (*v3.6+*) Enable allows to publish a catalog externally to make its vApp templates and media files available for subscription by organizations outside the vCloud Director installation. Default is `false`.
```
[the remark this answers is not in the document]
Comment (vbauzysvmware) [reply website/docs/d/catalog.html.markdown]: changed

File: website/docs/r/catalog.html.markdown (hunk 4 above)
Comment (vbauzysvmware) [reply website/docs/r/catalog.html.markdown:61]: removed

**Review by lvirbalas — APPROVED** [review]

**Pull request merged**